internal/coreinternal/pdatautil/map.go

```diff
@@ -0,0 +1,104 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pdatautil // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal/pdatautil"
+
+import (
+	"encoding/binary"
+	"hash"
+	"math"
+	"sort"
+
+	"github.com/cespare/xxhash/v2"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+)
+
+var (
+	valEmpty        = []byte{'\xf5'}
+	valBytesPrefix  = []byte{'\xf6'}
+	valStrPrefix    = []byte{'\xf7'}
+	valBoolTrue     = []byte{'\xf8'}
+	valBoolFalse    = []byte{'\xf9'}
+	valIntPrefix    = []byte{'\xfa'}
+	valDoublePrefix = []byte{'\xfb'}
+	valMapPrefix    = []byte{'\xfc'}
+	valMapSuffix    = []byte{'\xfd'}
+	valSlicePrefix  = []byte{'\xfe'}
+	valSliceSuffix  = []byte{'\xff'}
+)
+
+// MapHash return a hash for the provided map.
+// Maps with the same underlying key/value pairs in different order produce the same deterministic hash value.
+func MapHash(m pcommon.Map) uint64 {
```

Comment:
I would go with 16 bytes array (not slice) as return type. Allows to use UUID or other implementation.

Reply:
What I liked about uint64 is that results can be easily comparable but that maybe doesn't have colossal collision tolerance

Reply:
Changed to [16]byte